includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -772,15 +775,33 @@ protected function get_wp_query_args( $query_vars ) {
 			}
 		}
 
-		// Add the 'wc-' prefix to status if needed.
 		if ( ! empty( $query_vars['post_status'] ) ) {
+			// Add the 'wc-' prefix to status if needed.
 			if ( is_array( $query_vars['post_status'] ) ) {
 				foreach ( $query_vars['post_status'] as &$status ) {
 					$status = wc_is_order_status( 'wc-' . $status ) ? 'wc-' . $status : $status;
 				}
 			} else {
 				$query_vars['post_status'] = wc_is_order_status( 'wc-' . $query_vars['post_status'] ) ? 'wc-' . $query_vars['post_status'] : $query_vars['post_status'];
 			}
+
+			// Verify post_status is actually registered.
+			$post_stati = is_array( $query_vars['post_status'] ) ? $query_vars['post_status'] : array( $query_vars['post_status'] );
+			foreach ( $post_stati as $post_status ) {
+				$post_status_exists = get_post_stati( array( 'name' => $post_status ) );
+				if ( empty( $post_status_exists ) ) {
```

Comment:
should we collect these statuses in an array and then in the error message show all of them together (in case there are more than one un-registered statuses)?

Reply:
Ya that sounds like a good improvement.